src/main/scala/lenthall/ErrorOr.scala

```diff
@@ -0,0 +1,22 @@
+package lenthall
+
+import cats.data.Validated.{Invalid, Valid}
+import cats.data.{NonEmptyList, Validated}
+
+object ErrorOr {
+  type ErrorOr[A] = Validated[NonEmptyList[String], A]
+
+  implicit class FlatMap[A](val fa: ErrorOr[A]) extends AnyVal {
+    /**
+      * Not consistent with `Applicative#ap` but useful in for comprehensions.
+      *
+      * @see http://typelevel.org/cats/tut/validated.html#of-flatmaps-and-xors
+      */
+    def flatMap[B](f: A => ErrorOr[B]): ErrorOr[B] = {
```

Comment:
ToL: (aka my 2¢) The GCS filesystem does not need to know about Workflows. cromwell.core seems like it contains a lot of common code for a workflow engine:

- `ExecutionStatus`
- `OutputStore`
- `DockerCredentials`
- `WdlValueSimpleton`
- etc.

Perfect for "a catch-all for things which should be shared across subprojects"… again of a workflow engine.

Meanwhile, lenthall is currently common utilities that _may be_ applied outside of a workflow engine, with artifact dependencies marked as `provided`:
- `AggregatedException`
- `SwaggerUiHttpService`
- `IntegerUtil`
- etc.

`ErrorOr` and it's custom flatMap to me fits this description. Side 2¢ (for free) I would give the FlatMap implicit a better name to define if it was error-accumulation and/or short-circuiting.

A sign that the dependencies are busted would be if the GCS file system artifact has a non-"provided" dependency on wdl4s when running [`sbt gcsFileSystem/dependencyGraph`](https://github.com/jrudolph/sbt-dependency-graph)